osquery/events/windows/etw/etw_publisher_processes.cpp

```diff
@@ -0,0 +1,552 @@
+
+  // Userspace ETW Provider configuration
+  EtwProviderConfig userEtwProviderConfig;
+  userEtwProviderConfig.setName("Microsoft-Windows-Kernel-Process");
+  userEtwProviderConfig.setAnyBitmask(processStartStopKeyword);
+  userEtwProviderConfig.setPreProcessor(getPreProcessorCallback());
+  userEtwProviderConfig.setPostProcessor(getPostProcessorCallback());
+  userEtwProviderConfig.addEventTypeToHandle(EtwEventType::ProcessStart);
+  userEtwProviderConfig.addEventTypeToHandle(EtwEventType::ProcessStop);
+
+  // Adding the provider to the ETW Engine
+  Status userProviderAddStatus = EtwEngine().addProvider(userEtwProviderConfig);
+  if (!userProviderAddStatus.ok()) {
+    return userProviderAddStatus;
+  }
+
+  // Kernelspace ETW Provider configuration
+  EtwProviderConfig kernelProviderCfg;
+  kernelProviderCfg.setKernelProviderType(
+      EtwProviderConfig::EtwKernelProviderType::Process);
+  kernelProviderCfg.setPreProcessor(getPreProcessorCallback());
+  kernelProviderCfg.setPostProcessor(getPostProcessorCallback());
+  kernelProviderCfg.addEventTypeToHandle(EtwEventType::ProcessStart);
+  kernelProviderCfg.addEventTypeToHandle(EtwEventType::ProcessStop);
+
+  // Adding the provider to the ETW Engine
+  Status kernelProviderAddStatus = EtwEngine().addProvider(kernelProviderCfg);
+  if (!kernelProviderAddStatus.ok()) {
+    return kernelProviderAddStatus;
+  }
+
+  return Status::success();
+}
+
+// Callback to perform pre-processing logic
+void EtwPublisherProcesses::providerPreProcessor(
+    const EVENT_RECORD& rawEvent, const krabs::trace_context& traceCtx) {
+  // Helper accessors for userspace events
+  const EVENT_HEADER& eventHeader = rawEvent.EventHeader;
+  const unsigned int eventVersion = eventHeader.EventDescriptor.Version;
+
+  // Checking if new event is a supported one
+  if (!isSupportedEvent(eventHeader)) {
+    return;
+  }
+
+  // ETW event schema parsing
+  krabs::schema schema(rawEvent, traceCtx.schema_locator);
+  krabs::parser parser(schema);
+
+  // Internal ETW Event allocation - This event will be populated and dispatched
+  std::shared_ptr<EtwEventData> newEvent = std::make_shared<EtwEventData>();
+  if (newEvent == nullptr) {
+    LOG(WARNING) << "Cannot allocate new EtwEventData event";
+    return;
+  }
+
+  // Parsing ETW Event payload based on its type
+  bool eventShouldBeDispatched = false;
+  if (isSupportedKernelEvent(eventHeader)) {
+    // This is an event arriving from the ETW kernel provider!
+
+    // Opcode == Event ID
+    UCHAR opcode = eventHeader.EventDescriptor.Opcode;
+
+    if (opcode == etwProcessStartID) {
+      // Checking event opcode to determine if this is an event containing
+      // process-start information
+
+      // Event type initialization
+      newEvent->Header.Type = EtwEventType::ProcessStart;
+
+      // Allocating process-start specific payload
+      auto procStartData = std::make_shared<EtwProcessStartData>();
+      if (!procStartData) {
+        return;
```

Comment:
log?

Reply:
Fixed, on this case it worth logging the error because we are sure this is an event we are interested in